include/uapi/linux/vhost.h

```diff
@@ -114,6 +116,6 @@
 /* VHOST_VSOCK specific defines */
 
 #define VHOST_VSOCK_SET_GUEST_CID	_IOW(VHOST_VIRTIO, 0x60, __u64)
-#define VHOST_VSOCK_SET_RUNNING		_IOW(VHOST_VIRTIO, 0x61, int)
+#define VHOST_VSOCK_SET_RUNNING		VHOST_SET_RUNNING
```

Comment:
No, it isn't a no-op. I'll explain this in the design document.

Reply:
@plbossart from the document (an older version, before corrections, it's the very last paragraph of the entire document):
> Guest reboot also requires special handling to deinitialise virtual queues. At the moment there is no standard way to get notified about the guest reboot event on the host. Various vhost drivers implement this in their own context-specific ways. E.g. the vhost networking driver uses a link status update for that purpose. However the vsock vhost driver comes very close to our needs. It doesn't have a suitable context notification, therefore it implements a dedicated QEMU / misc-device ioctl VHOST_VSOCK_SET_RUNNING. We reuse the ioctl
definition, but change its name to VHOST_SET_RUNNING. This way no existing user-space software has to be modified. Then we use this ioctl() in QEMU SOF code to inform the SOF vhost driver about a guest reboot. This is then used to reset VirtQueue status on the host.

Reply:
sorry, this is awfully convoluted.

So the vsock host driver will now handle the VSOCK_SET_RUNNING ioctl
and the SOF host driver will also handle the VSOC_SET_RUNNING iotcl.

How do you know when an ioctl is used for one or the other case?

Reply:
Mmh, I'm not sure I understand your question. You don't need to know, there aren't two cases, this is just one case. It is just one ioctl(), no driver needs to handle it in two different ways. I just say - this ioctl that vsock is using, is useful for others too, let's make it generic. But because this is user-space ABI, we cannot just rename it if we want to let Linus sleep well and want to avoid his now oh so mild and civilised critics. So, we don't change anything, legacy software still can happily use the old name and legacy binary software will work too, because we also don't change the numeric value. We just add an alias, because it would be silly for unrelated drivers to use a ...VSOCK... ioctl. And hopefully with time vsock can also transition to the generic name and then the original one can be deprecated and eventually removed.

Reply:
ok, maybe I was reading too much in the explanation "This patch makes that ioctl generic"

It's just a rename with backwards compatibility, there is actually no functionality added.